conjure-http/src/client.rs

```diff
@@ -223,3 +237,219 @@ pub trait AsyncWriteBody<W> {
     where
         W: 'async_trait;
 }
+
+/// A trait implemented by request body serializers used by custom Conjure client trait
+/// implementations.
+pub trait SerializeRequest<'a, T, W> {
+    /// Returns the body's content type.
+    fn content_type(value: &T) -> HeaderValue;
+
+    /// Returns the body's length, if known.
+    ///
+    /// Empty and fixed size bodies will have their content length filled in automatically.
+    ///
+    /// The default implementation returns `None`.
+    fn content_length(value: &T) -> Option<u64> {
+        let _value = value;
+        None
+    }
+
+    /// Serializes the body.
+    fn serialize(value: T) -> Result<RequestBody<'a, W>, Error>;
+}
+
+/// A trait implemented by request body serializers used by custom async Conjure client trait
+/// implementations.
+pub trait AsyncSerializeRequest<'a, T, W> {
+    /// Returns the body's content type.
+    fn content_type(value: &T) -> HeaderValue;
+
+    /// Returns the body's length, if known.
+    ///
+    /// Empty and fixed size bodies will have their content length filled in automatically.
+    ///
+    /// The default implementation returns `None`.
+    fn content_length(value: &T) -> Option<u64> {
+        let _value = value;
+        None
+    }
+
+    /// Serializes the body.
+    fn serialize(value: T) -> Result<AsyncRequestBody<'a, W>, Error>;
+}
+
+/// A body serializer which encodes the type into JSON.
+///
+/// # Note
+///
+/// The default `serde_json` serializer is used, *not* the `conjure-serde` serializer.
```

Comment:
Does this seem like the right approach? We may want a `ConjureRequestSerializer` or something like that for things that talk to non-Conjure WC resources that uses our conjure serialization behavior?

Reply:
nit: Maybe `SerdeRequestSerializer` to make it more explicit?

Reply:
We definitely need to mention that it's JSON in the name. SerdeJsonRequestSerializer is more specific but maybe a bit wordy?

Reply:
I think I'm going to unify the standard Conjure codegen to use these same traits, so I think I'll actually just replace this with one that does the standard Conjure thing.

Reply:
Updated